azure/virtual-machine/windows.tf

```diff
@@ -0,0 +1,58 @@
+resource "azurerm_windows_virtual_machine" "main" {
+  count = local.is_windows ? 1 : 0
+
+  name                = var.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+
+  computer_name  = var.computer_name == null ? var.name : var.computer_name
+  admin_username = var.username
+  admin_password = var.password
+
+  availability_set_id   = var.availability_set_id
+  size                  = var.size
+  network_interface_ids = [azurerm_network_interface.main.id]
+  boot_diagnostics {}
```

Comment:
whilst rarely used, would it make sense to pass in `storage_account_uri` into here just in case it's needed one day?

Reply:
Added for linux and windows.